apps/api/src/app/bridge/usecases/sync/sync.usecase.ts

```diff
@@ -90,6 +96,47 @@ export class Sync {
     return persistedWorkflowsInBridge;
   }
 
+  private async invalidateStaleWorkflows(command: SyncCommand, discover: DiscoverOutput) {
+    const existingWorkflows = await this.notificationTemplateRepository.find(
+      {
+        _environmentId: command.environmentId,
+        type: {
+          $in: [WorkflowTypeEnum.ECHO, WorkflowTypeEnum.BRIDGE],
+        },
+      },
+      {
+        projection: {
+          'triggers.0.identifier': 1,
+          _id: 1,
+        },
+      }
+    );
+    const discoverWorkflowIdentifiers = discover.workflows.map((workflow) => workflow.workflowId);
+    const staleWorkflows = existingWorkflows.filter(
+      (_worklfow) => !discoverWorkflowIdentifiers.includes(_worklfow.triggers[0].identifier)
+    );
+    const staleWorkflowIdentifiers = staleWorkflows.map((workflow) => workflow.triggers[0].identifier);
+    const staleWorkflowIds = staleWorkflows.map((workflow) => workflow._id);
+
+    for (const identifier of staleWorkflowIdentifiers) {
+      await this.invalidateCacheService.invalidateByKey({
+        key: buildNotificationTemplateIdentifierKey({
```

Comment:
The update-workflow.command that is handling existing workflows already invalidates the cache for both keys - See [here](https://github.com/novuhq/novu/blob/8e35dad2c5675301d67b4cbfb7ff9af3eb001137/libs/application-generic/src/usecases/workflow/update-workflow/update-workflow.usecase.ts#L212).

I suggest we apply a more surgical solution to avoid fetching working multiple times from the database that adds extra DB requests.

Reply:
You make a good point. The data that gets validated on updates should indeed be part of what we check. 

To illustrate today's issue, here’s the scenario we need to address:

1. A trigger is sent to a non-existent workflow with ID "best-workflow." On the trigger, we try to fetch the workflow, but since it doesn’t exist, we store `null` in the cache.
2. We then sync "best-workflow."
3. When we trigger "best-workflow" again, we pull the `null` value from the cache.

To resolve this, [we could prevent `null` from being cached on creation](https://github.com/novuhq/novu/pull/6887/commits/85a1385d044795f3eaedf9432d762dcb6220b698).

While reviewing the issue, I thought we had another problem, but I realized it's not a bug—just an instance of stale memory that we could release. Now, I’m considering [removing the extra cache invalidation step](https://github.com/novuhq/novu/pull/6887/commits/c0d66c3a3df679ab6938b49ffd923d9c157aa5a6) and relying on the TTL to clear it, which would simplify the logic and reduce the load on Redis. 

What do you think?